table/table_reader_caller.h

```diff
@@ -0,0 +1,37 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+namespace rocksdb {
+// A list of callers for a table reader. It is used to trace the caller that
+// accesses on a block. This is only used for block cache analysis.
+enum TableReaderCaller : char {
```

Comment:
Are all these necessary? Do we want to distinguish between 'SSTFileReader', 'repair', 'unit test', etc.?

Reply:
I agree that we may not need to differentiate these callers. 

I originally only want to add compactionRefill and SSTIngestion. As I changed the NewIterator in table reader, I realize many other places touch the NewIterator. So later I put these callers in as they can be differentiated easily.

Reply:
Sure. I added a kUncategorized enum and comments inline. "A user may use kUncategorized if the caller is not useful for analysis or the table reader is called in the test environment, e.g., unit test, table reader benchmark, etc."